Prefer usage of `StandardCharsets.UTF_8` over `"UTF-8"` charset reference

thomasdarimont opened the pull request:
Used StandardCharsets.UTF_8 instead of "UTF-8" whenever possible. This reduces some warnings, simplifies the code and is more readable.

Fixes #35080

<!---
Please read https://github.com/keycloak/keycloak/blob/main/CONTRIBUTING.md and follow these guidelines when contributing to Keycloak
-->

thomasdarimont committed:
Prefer usage of StandardCharsets.UTF_8 over "UTF-8" charset reference (#35080)

Used StandardCharsets.UTF_8 instead of "UTF-8" whenever possible.
This reduces some warnings, simplifies the code and is more readable.

Fixes #35080

Signed-off-by: Thomas Darimont <thomas.darimont@googlemail.com>

jonkoops reviewed (dismissed):
Nice work! I appreciate the fact that we don't have to handle a bunch of unnecessary `UnsupportedEncodingException`s anymore.

jonkoops commented:
Looks like the tests are not happy though, so this probably needs a couple of passes before it is mergable.

thomasdarimont committed:
Must keep using the string charset reference in BasicAuthHelper

The core module is still compiled with Java 8 compatibility, hence we
need to use the old UTF-8 reference.

Signed-off-by: Thomas Darimont <thomas.darimont@googlemail.com>

thomasdarimont committed:
Fix compilation errors due to unused exception handlers

Fixes #35080

Signed-off-by: Thomas Darimont <thomas.darimont@googlemail.com>

ahus1 committed:
Remove unused imports

Fixes #35080

Signed-off-by: Alexander Schwartz <aschwart@redhat.com>
